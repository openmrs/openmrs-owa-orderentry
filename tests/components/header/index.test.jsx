import React from 'react';
import expect from 'expect';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import ConnectedHeader, {
  Header
} from '../../../app/js/components/header';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Header {...props} />);
  }
  return mountedComponent;
};

describe('Component: Header', () => {
  beforeEach(() => {
    props = {
      fetchCurrentSession: jest.fn(),
      fetchLocations: jest.fn(),
      currentLocation: {},
      locations: [],
      setCurrentLocation: jest.fn(),
    };
    mountedComponent = undefined;
  });

  it('should render properly', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  describe('Connected Header component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        sessionReducer: {
          currentLocation: { display: "Amani Hospital" },
          currentUser: "admin",
        },
        locationReducer: {
          locationTags: []
        }
      });
      const wrapper = shallow(<ConnectedHeader store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
