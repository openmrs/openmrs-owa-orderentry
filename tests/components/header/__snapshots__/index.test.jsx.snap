// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentLocation={Object {}}
    currentUser=""
    fetchCurrentSession={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchLocations={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    locations={Array []}
    setCurrentLocation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <View
        currentLocation={Object {}}
        currentUser=""
        locationDropdown={false}
        locations={Array []}
        setCurrentLocation={[MockFunction]}
        toggleState={[Function]}
        userDropdown={false}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "currentLocation": Object {},
        "currentUser": "",
        "locationDropdown": false,
        "locations": Array [],
        "setCurrentLocation": [MockFunction],
        "toggleState": [Function],
        "userDropdown": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <View
          currentLocation={Object {}}
          currentUser=""
          locationDropdown={false}
          locations={Array []}
          setCurrentLocation={[MockFunction]}
          toggleState={[Function]}
          userDropdown={false}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentLocation": Object {},
          "currentUser": "",
          "locationDropdown": false,
          "locations": Array [],
          "setCurrentLocation": [MockFunction],
          "toggleState": [Function],
          "userDropdown": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
